scripts/spcgeonode/docker-compose.yml

```diff
@@ -22,10 +22,25 @@ x-common-django:
     - TIME_ZONE=${TIME_ZONE}
     - ALLOWED_HOSTS=['nginx','127.0.0.1','localhost','$HTTPS_HOST','$HTTP_HOST']
     - SECRET_KEY=${SECRET_KEY}
+
+    ## UNCOMMENT FOR SSL
+    #- GEOSERVER_PUBLIC_LOCATION=https://${HTTPS_HOST}/gs/
+    #- GEOSERVER_WEB_UI_LOCATION=https://${HTTPS_HOST}/geoserver/
+    #- SITEURL=https://${HTTPS_HOST}
+    ## COMMENT FOR SSL
+    - GEOSERVER_PUBLIC_LOCATION=http://${HTTP_HOST}/gs/
+    - GEOSERVER_WEB_UI_LOCATION=http://${HTTP_HOST}/geoserver/
+    - SITEURL=http://${HTTP_HOST}

```

Comment:
This improves the current situation, but it would be a different way to specify URL (using full URLs instead of HTTP{,S}_{HOST_PORT}. I prefer your approach actually, but it would be nice to avoid mixing things together.

Reply:
So what exactly is your proposal?

In theory HTTP and HTTPS Host are equal (lets say geonode.domain.com). What is different is the protocoll. One solution could be to use something like:

`GEOSERVER_PUBLIC_LOCATION=${PROTOCOLL}://${HOST}/gs/`

and where HTTP is needed (container to container communication:
`GEOSERVER_LOCATION=http://nginx/geoserver/`

The default geonode composition does use some shell scripting with `if ... fi` blocks to concatenate URLs.

Reply:
I would say that we can export those variables in the entrypoint, based on the if/fi block you are mentioning.

Reply:
Will be set in django/docker-entrypoint.sh